mitmproxy/protocol/http.py

```diff
@@ -187,8 +187,7 @@ def __call__(self):
                     flow.request.set_auth(self.config.upstream_auth)
                 self.process_request_hook(flow)
 
-                if not flow.response:
-                    self.establish_server_connection(flow)
+                if not flow.response and self.establish_server_connection(flow):
```

Comment:
We can do that. Raising or returning False will have the same effect - it needs to be handled. Right now, nobody cares if it fails.
Do you prefer an exception in this case?